ananse/network.py

```diff
@@ -196,13 +145,31 @@ def get_gene_dataframe(self, peak_bed, up=100000, down=100000):
             ["gene", "loc", "dist"]
         ]
 
-        p = p[p["dist"] < up - 1]
         # remove distance more than 100k interaction, for weight calculate
-        p.gene = [i.upper() for i in list(p.gene)]
+        p = p[p["dist"] < up - 1]
+
+        weight = self.distance_weight(
+            include_promoter=self.include_promoter,
+            include_enhancer=self.include_enhancer,
+            remove=promoter,
+            keep1=full_weight_region,
+        ).set_index("dist")
+        p = p.join(weight, on="dist")
 
+        p.gene = [i.upper() for i in list(p.gene)]
```

Comment:
I'm btw not a fan of just changing gene names into upper or lower. But this has to be changed simultaneously in binding as well

Reply:
in binding, TF factors were (and still are) also converted to upper case

Reply:
No indeed! This is left-over. We should not need to touch gene names in this manner.